Add Subheader widget style

pardom-zz opened the pull request:
Not sure about how you want to approach this, but I recommend that widget styles be added so that individual attributes don't need to be applied to every subheader or other widget.

pardom committed:
Add Subheader widget style

Not sure about how you want to approach this, but I recommend that widget styles be added so that individual attributes don't need to be applied to every subheader or other widget.

AoDevBlue commented:
That's interesting to apply these values faster, thank you. I will integrate it after small modifications.

I agree to introduce small styles whenever it's possible to simplify usage, but it also raises an interesting question about the scope of the library.

Because MaterialValues contains a lot of the guidelines values, it would be possible to create styles and themes for the components, in the spirit of the official Material theme of the platform (for example, [themes_material.xml](https://chromium.googlesource.com/android_tools/+/master/sdk/platforms/android-24/data/res/values/themes_material.xml) and [styles.xml](https://chromium.googlesource.com/android_tools/+/master/sdk/platforms/android-24/data/res/values/styles.xml)). But I don't currently have the insight needed to do that, and maybe it should be in another library.

So, as for now, I won't add widget styles for all components. However, if somebody thinks a style would be great somewhere, feel free to create a PR!

pardom committed:
Fix subheading size, color, and font family

Specs taken from here: https://material.google.com/components/subheaders.html#subheaders-list-subheaders

pardom committed:
Use textAppearance

AoDevBlue commented:
I can't accept 3b790d2, because the values of the [typography's subheading](https://material.google.com/style/typography.html) are not linked to the [subheader component](https://material.google.com/components/subheaders.html).

Beside, typography styles are just a shorthand for size and font family, they should be color independent.
Both light themes and dark themes should be possible using MaterialValues, but currently it misses some dark themes values as they are missing from the guidelines as well.

For the subheader, for example, the default text color should be specified to `@color/material_typography_secondary_text_color_dark` for light themes and `@color/material_typography_secondary_text_color_light` for dark themes. That's where using custom themes would be super useful, as we can reference the `textColorSecondary` attribute.
